bravado/http_client.py

```diff
@@ -197,6 +197,9 @@ def __init__(self, session, request):
         self.session = session
         self.request = request
 
+    def result(self, *args, **kwargs):
```

Comment:
The interface definition is not really clear in the current version (it's more of a base class). I think the this interface needs a lot of work, it doesn't have to happen right now (but it should happen before the bravado release).

A major problem with the current implementation is that it's mutable. There is no reason for it to be.

I think it should be something like this:

```python

RequestParams = namedtuple('RequestParams', ...)

class HttpClient(object):

    # this would replace start_request()
    def request(self, request_params):
        """Returns an HttpFuture"""

class HttpFuture(object):

    def result(self, timeout=DEFAULT_TIMEOUT):
        """Returns a swagger response, or raises an exception if there was an error making the request"""

```